---
title: "R Notebook"
output: html_notebook
---
**include dependencies**
```{r}
library(data.table)
library(dplyr)
library(purrr)
library(DESeq2)
```

#### Data Loading
*Only use relevant columns*
```{r}
infected_raw <- fread(file="data/GSE162323_slam_inf_params.txt", check.names=TRUE)
uninfected_raw <- fread("data/GSE162323_slam_uninf_params.txt", check.names=TRUE)
infected_1 = infected_raw[,c(1, 3:8)]
uninfected_1 = uninfected_raw[,c(1, 3:8)]
```

*Remove duplicate rows with median and merge uninfected and infected gene expression datasets*
```{r}
infected_dedup <- infected_1 %>%
  group_by(gene.symbol) %>%
  summarize(
    one.hour.inf = as.integer(median(X1h.4sU.Readcount, na.rm=TRUE)),
    two.hour.inf = as.integer(median(X2h.4sU.rep1.Readcount)),
    three.hour.inf = as.integer(median(X3h.4sU.Readcount)),
    four.hour.inf = as.integer(median(X4h.4sU.Readcount))
    )

uninfected_dedup <- uninfected_1 %>%
  group_by(gene.symbol) %>%
  summarize(
    one.hour.uninf = as.integer(median(X1h.4sU.Readcount, na.rm=TRUE)),
    two.hour.uninf = as.integer(median(X2h.4sU.rep2.Readcount)),
    three.hour.uninf = as.integer(median(X3h.4sU.Readcount)),
    four.hour.uninf = as.integer(median(X4h.4sU.Readcount))
    )

cts_merged <- merge(x=infected_dedup, y=uninfected_dedup, by="gene.symbol")
cts <- cts_merged[,-1]
rownames(cts) <- cts_merged[,1]
print(cts)
```
#In this results table, we used the wrapper function DESeq() to do the differential expression analysis. We also used tidyverse to do sort or filter before saving our results to a file since it is not sorted or filtered. Finally, we printed out the top results. An interesting result is that LCK and CD7 had similar log2FoldChange results.

#### Per Gene Expression Range
```{r}
diff <- cts |>
  apply(1, function(r) log(max(r) - min(r), 2))

hist(diff, ylab = "Log2Frequency")
```
### PCA Plot
*Coldata*
```{r}
coldata <- as.data.frame(names(cts))
coldata$condition <- c("infected", "infected", "infected", "infected", "uninfected", "uninfected", "uninfected", "uninfected")
coldata$batches <-  c("one", "two", "three", "four", "one", "two", "three", "four");
```

*Gene Expression Matrix*
```{r}
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ condition + batches)
dds <- DESeq(dds) #this is doing a lot of number crunching
res <- results(dds)
vsd <- vst(dds, blind=FALSE)
res
```
*PCA Plot TODO: Correct axis (infected vs uninfected)*
```{R}
plotPCA(vsd, intgroup=c("condition"))
```
#In this plot, we plotted rlog transformed data to check results. To compare the infected and uninfected genes, we plotted one with plotPCA() function. An interesting result is that the plot produced huge and distinct differences between infected and uninfected genes. 

### Volcano Plot
#### Some Extra Analysis
```{r}
set.seed(9823249)
deseq_df <- res |>
  as.data.frame() |>
  tibble::rownames_to_column("Gene") |>
  dplyr::mutate(threshold = padj < 0.05) |>
  arrange(desc(log2FoldChange))

head(deseq_df)
```
```{r}
volcano_plot <- EnhancedVolcano::EnhancedVolcano(
  deseq_df,
  lab = deseq_df$Gene,
  x = "log2FoldChange",
  y = "padj",
  pCutoff = 0.01
)
volcano_plot
```
#In the volcano plot, we plotted the Log2FoldChange on the x-axis and padj on the y-axis. The padj variable is the p values corrected with Benjamini-Hochberg. An interesting result is that pCutoff argument can be made 0.01, which the default value is 1e-05, since we were using adjusted p values.
