---
title: "R Notebook"
output:
  word_document: default
  html_notebook: default
---
**include dependencies**
```{r, echo=FALSE}
library(data.table)
library(dplyr)
library(purrr)
library(DESeq2)
library(ComplexHeatmap)
library(clusterProfiler)
library(DOSE)
library(org.Hs.eg.db)
```

#### Data Loading
*Only use relevant columns*
```{r}
infected_raw <- fread(file="data/GSE162323_slam_inf_params.txt", check.names=TRUE)
uninfected_raw <- fread("data/GSE162323_slam_uninf_params.txt", check.names=TRUE)
infected_1 = infected_raw[,c(1, 3:8)]
uninfected_1 = uninfected_raw[,c(1, 3:8)]
head(infected_raw)
```

*Remove duplicate rows with median and merge uninfected and infected gene expression datasets*
```{r}
infected_dedup <- infected_1 %>%
  group_by(gene.symbol) %>%
  summarize(
    one.hour.inf = as.integer(median(X1h.4sU.Readcount, na.rm=TRUE)),
    two.hour.inf.rep1 = as.integer(median(X2h.4sU.rep1.Readcount)),
    two.hour.inf.rep2 = as.integer(median(X2h.4sU.rep2.Readcount)),
    three.hour.inf = as.integer(median(X3h.4sU.Readcount)),
    four.hour.inf = as.integer(median(X4h.4sU.Readcount))
    )

uninfected_dedup <- uninfected_1 %>%
  group_by(gene.symbol) %>%
  summarize(
    one.hour.uninf = as.integer(median(X1h.4sU.Readcount, na.rm=TRUE)),
    two.hour.uninf.rep1 = as.integer(median(X2h.4sU.rep1.Readcount)),
    two.hour.uninf.rep2 = as.integer(median(X2h.4sU.rep2.Readcount)),
    three.hour.uninf = as.integer(median(X3h.4sU.Readcount)),
    four.hour.uninf = as.integer(median(X4h.4sU.Readcount))
    )

cts_merged <- merge(x=infected_dedup, y=uninfected_dedup, by="gene.symbol")
cts <- cts_merged[,-1]
rownames(cts) <- cts_merged[,1]
print(cts)
```
<<<<<<< HEAD
=======
#In this results table, we used the wrapper function DESeq() to do the differential expression analysis. We also used tidyverse to do sort or filter before saving our results to a file since it is not sorted or filtered. Finally, we printed out the top results. An interesting result is that LCK and CD7 had similar log2FoldChange results.

#### Per Gene Expression Range
```{r}
diff <- cts |>
  apply(1, function(r) log(max(r) - min(r), 2))

hist(diff, ylab = "Log2Frequency")
```
>>>>>>> a1eabce43919f60f8e103bbac9d00764427238a6
### PCA Plot
*Coldata*

```{r}
coldata <- as.data.frame(names(cts))
coldata$condition <- c("infected", "infected", "infected", "infected", "infected", "uninfected", "uninfected", "uninfected", "uninfected", "uninfected")
coldata$batches <-  c("one", "two.rep1", "two.rep2", "three", "four", "one","two.rep1", "two.rep2", "three", "four");
```

*Gene Expression Matrix*
```{r}
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ condition + batches)
dds <- DESeq(dds) #this is doing a lot of number crunching
res <- results(dds)
vsd <- vst(dds, blind=FALSE)
head(res)
```
*PCA Plot TODO: Correct axis (infected vs uninfected)*
```{R}
plotPCA(vsd, intgroup=c("condition"))
```
#In this plot, we plotted rlog transformed data to check results. To compare the infected and uninfected genes, we plotted one with plotPCA() function. An interesting result is that the plot produced huge and distinct differences between infected and uninfected genes. 

### Volcano Plot
#### Some Extra Analysis
```{r}
set.seed(96946461)
deseq_df <- res |>
  as.data.frame() |>
  tibble::rownames_to_column("Gene") |>
  dplyr::mutate(threshold = padj < 0.01) |>
  arrange(desc(log2FoldChange))

head(deseq_df, n=10)
```
```{r}
volcano_plot <- EnhancedVolcano::EnhancedVolcano(
  deseq_df,
  lab = deseq_df$Gene,
  x = "log2FoldChange",
  y = "padj",
  pCutoff = 0.01
)
volcano_plot
```
### clustProfiler GO
```{r}
geneList <- deseq_df[,3]
names(geneList) = deseq_df[,1]
go_res <- gseGO(
  geneList = sort(geneList, decreasing = TRUE),
  OrgDb = org.Hs.eg.db,
  ont = "CC",
  keyType = 'SYMBOL',
  pvalueCutoff = 0.01,
  verbose = FALSE
)

head(go_res, n=10)

```
#### Dot Plot
```{r}
dotplot(go_res, showCategory=10)
```
#### CNet Plot
```{r}
cnetplot(go_res, node_label="category", 
        cex_label_category = 1.2) 
```

For the gene ontology using clusterProfiler we looked at the biological processes domain. Many of the genes that are being expressed are related to the immune response.

### clustProfiler Disease Ontology


```{r}
geneListEntrez <- deseq_df[,3]
names(geneListEntrez) <- as.character(deseq_df[,1])
y <- gseDO(sort(geneListEntrez, decreasing = TRUE),
           pvalueCutoff  = 0.05,
           pAdjustMethod = "BH",
           keyType = 'SYMBOL',
           verbose       = FALSE
           )

```
```{r}
dotplot(y)
```
